homeassistant/components/bitvavo/__init__.py

```diff
@@ -0,0 +1,218 @@
+"""Gather the market details from Bitvavo."""
+from __future__ import annotations
+
+import logging
+
+from bitvavo.BitvavoClient import BitvavoClient
+from bitvavo.BitvavoExceptions import BitvavoException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    ASSET_VALUE_BASE,
+    ASSET_VALUE_CURRENCIES,
+    CONF_API_SECRET,
+    CONF_MARKETS,
+    CONF_SHOW_EMPTY_ASSETS,
+    DOMAIN,
+    PLATFORMS,
+    SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Bitvavo from a config entry."""
+    api_key = entry.data[CONF_API_KEY]
+    api_secret = entry.data[CONF_API_SECRET]
+    markets = entry.data[CONF_MARKETS]
+    show_empty_assets = entry.options.get(CONF_SHOW_EMPTY_ASSETS, True)
+
+    try:
+        client = BitvavoClient(api_key, api_secret)
```

Comment:
Constructing the client can not raise `BitvavoException` ?
<https://github.com/cyberjunky/bitvavo-aio/blob/main/bitvavo/BitvavoClient.py#L27>

Reply:
No the init doesn't run anything yet.